### PR TITLE
persist /etc/rancher/node across reboots

### LINKED ISSUE #289: kubectl: failed to find Session for client (after reboot)
seeing errors invoking `kubectl` after rebooting:
```
k3os-4224 [~]$ kubectl -n kube-system logs -l k8s-app=kube-dns
Error from server: Get https://k3os-4224:10250/containerLogs/kube-system/coredns-d798c9dd-ck27h/coredns?tailLines=10: failed to find Session for client k3os-4224
k3os-4224 [~]$ 
```

with noise in the `/var/log/k3s-service.log`:
```
2019-11-20 17:54:28.674346 I | http: TLS handshake error from 127.0.0.1:60264: remote error: tls: bad certificate
time="2019-11-20T17:54:28.685803956Z" level=error msg="Node password validation failed for 'k3os-4224', using passwd file '/var/lib/rancher/k3s/server/cred/node-passwd'"
time="2019-11-20T17:54:28.685938289Z" level=error msg="Node password rejected, duplicate hostname or contents of '/etc/rancher/node/password' may not match server node-passwd entry, try enabling a unique node name with the --with-node-id flag"
time="2019-11-20T17:54:29.260505179Z" level=info msg="Waiting for master node  startup: resource name may not be empty"
time="2019-11-20T17:54:30.262136877Z" level=info msg="Waiting for master node  startup: resource name may not be empty"
time="2019-11-20T17:54:31.263759798Z" level=info msg="Waiting for master node  startup: resource name may not be empty"
W1120 17:54:31.713374    2072 garbagecollector.go:640] failed to discover some groups: map[metrics.k8s.io/v1beta1:the server is currently unable to handle the request]
time="2019-11-20T17:54:32.269654396Z" level=info msg="Waiting for master node  startup: resource name may not be empty"
time="2019-11-20T17:54:33.272338879Z" level=info msg="Waiting for master node  startup: resource name may not be empty"
2019-11-20 17:54:33.689656 I | http: TLS handshake error from 127.0.0.1:60278: remote error: tls: bad certificate
time="2019-11-20T17:54:33.700857057Z" level=error msg="Node password validation failed for 'k3os-4224', using passwd file '/var/lib/rancher/k3s/server/cred/node-passwd'"
time="2019-11-20T17:54:33.700990339Z" level=error msg="Node password rejected, duplicate hostname or contents of '/etc/rancher/node/password' may not match server node-passwd entry, try enabling a unique node name with the --with-node-id flag"
time="2019-11-20T17:54:34.273747121Z" level=info msg="Waiting for master node  startup: resource name may not be empty"
```

I think this is caused by `/etc/rancher` not being persistent across reboots.

### PR DESCRIPTION
**Pull request by dweomer**
Fixes #289

### CODE REVIEW

File: overlay/libexec/k3os/mode-local [hunk 1]
```diff
@@ -10,4 +10,12 @@ setup_ssh()
     ln -s /var/lib/rancher/k3os/ssh /etc/ssh
 }
 
+setup_rancher_node()
+{
+    mkdir -p /etc/rancher
+    mkdir -p /var/lib/rancher/k3os/node
```
Comment (erikwilson) [inline overlay/libexec/k3os/mode-local]:
```suggestion
    mkdir -p /var/lib/rancher/k3os/etc-rancher
```

The information that needs to be persisted is in `/etc/rancher/node/`, the sub-directory would then be `/var/lib/rancher/k3os/node/node` which seems a little weird, so maybe a good idea to make the parent directory name more descriptive of the use.

Comment (dweomer) [reply overlay/libexec/k3os/mode-local]: @erikwilson note the trailing slash: `ln -s /var/lib/rancher/k3os/node /etc/rancher/`

on a local instance:
```
k3os-27078 [~]$ grep VERSION_ID /etc/os-release 
VERSION_ID="7fa0889"
k3os-27078 [~]$ cat /run/k3os/mode 
local
k3os-27078 [~]$ realpath /etc/rancher/node/password
/var/lib/rancher/k3os/node/password
k3os-27078 [~]$ 
```

Comment (dweomer) [reply overlay/libexec/k3os/mode-local]: This kinda comes down to:
- do we want/need to persist all of `/etc/rancher`?
- or is `/etc/rancher/node` enough?

I'd like a flag in k3s where I can tell it where to persist and find this kind of information, but failing that I am okay with a little targeted symlink magic.

**Comment by erikwilson** [comment]
Yah, sorry, that makes sense. Not sure if we should persist all of
`/etc/rancher`, seems like we should, but dunno what k3os might overwrite.

/LGTM

**Pull request merged**